[WPE] Fix build with GLib older than 2.74

aperezdc opened the pull request:
#### 906329430eb27e345bf525433f393e8a3912936a
<pre>
[WPE] Fix build with GLib older than 2.74
<a href="https://bugs.webkit.org/show_bug.cgi?id=265286">https://bugs.webkit.org/show_bug.cgi?id=265286</a>

Unreviewed build fix.

While 271376@main attempted to fix g_io_unix_fd_list_get() being
undefined with versions of GLib older than 2.74, it still resulted
in build issues in some cases. This changes the included header
to the &lt;gio/gunixfdlist.h&gt; one listed in the documentation.

* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSessionLogind.cpp: Include
  gio/gunixfdlist.h as per the documentation instead of glib-unix.h.

Canonical link: <a href="https://commits.webkit.org/271511@main">https://commits.webkit.org/271511@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/df115905832f48228ef781c7b823cf2e5cca63a1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28649 "Failed to checkout and rebase branch from PR 21275") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/7292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/30022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/31276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/26149 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/9406 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/4655 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/31276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28919 "Failed to checkout and rebase branch from PR 21275") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/9406 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/14/builds/30022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/31276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/9406 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/30022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/32612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/9406 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/30022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/32612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/5354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/4655 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/32612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/6966 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/30022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/5820 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/3695 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/5873 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/WPEPlatform/CMakeLists.txt:
```diff
@@ -74,12 +74,14 @@ set(WPEPlatform_PRIVATE_INCLUDE_DIRECTORIES
 
 set(WPEPlatform_SYSTEM_INCLUDE_DIRECTORIES
     ${GLIB_INCLUDE_DIRS}
+    ${GIO_UNIX_INCLUDE_DIRS}
```
This was already added to the DRM makefile, no?

aperezdc replied:
I'll double check, I bet you are right and the change to use `<gio/gunixfdlist.h>` is enough.

aperezdc replied:
Checked, and indeed changing the include is enough. Patch updated.

carlosgcampos commented on Source/WebKit/WPEPlatform/CMakeLists.txt:
```diff
@@ -74,12 +74,14 @@ set(WPEPlatform_PRIVATE_INCLUDE_DIRECTORIES
 set(WPEPlatform_LIBRARIES
     WTF
     ${GLIB_GIO_LIBRARIES}
+    ${GIO_UNIX_LIBRARIES}
```
And this.

aperezdc committed:
[WPE] Fix build with GLib older than 2.74
https://bugs.webkit.org/show_bug.cgi?id=265286

Unreviewed build fix.

While 271376@main attempted to fix g_io_unix_fd_list_get() being
undefined with versions of GLib older than 2.74, it still resulted
in build issues in some cases. This changes the included header
to the <gio/gunixfdlist.h> one listed in the documentation.

* Source/WebKit/WPEPlatform/wpe/drm/WPEDRMSessionLogind.cpp: Include
  gio/gunixfdlist.h as per the documentation instead of glib-unix.h.

Canonical link: https://commits.webkit.org/271511@main
